Despair-Games/poketernity
# [Move] Fully implement Shell Side Arm

## What are the changes the user will see?

Shell side arm now makes contact if physical

## Why am I making these changes?

Copying more homework to remove (P)

## What are the changes from a developer perspective?

* Added doc to `ShellSideArmCategoryAttr`
* Simplified an if/elseif into an if with ||
* Added a `move.makesContact()` in the attr if it was physical 
* Updated test

## Screenshots/Videos

See PR 4982

## How to test the changes?

`npm run test shell_side_arm`
```
✓ |main| src/test/moves/shell_side_arm.test.ts (5) 7311ms
   ✓ Moves - Shell Side Arm (5) 7307ms
     ✓ becomes a physical attack if forecasted to deal more damage as physical 1413ms
     ✓ should make contact if the move becomes physical 1537ms
     ✓ remains a special attack if forecasted to deal more damage as special 1310ms
     ✓ should not make contact if the move becomes special 1357ms
     ✓ respects stat stage changes when forecasting base damage 1654ms
```

Overrides
```ts
STARTER_SPECIES_OVERRIDE: Species.CHARIZARD, // close attack and special attack, higher spA
STARTING_LEVEL_OVERRIDE: 100,
MOVESET_OVERRIDE: [Moves.SWORDS_DANCE, Moves.NASTY_PLOT, Moves.SHELL_SIDE_ARM], // to boost attack or special attack
OPP_SPECIES_OVERRIDE: Species.MEW, // equal def and spdef
OPP_LEVEL_OVERRIDE: 5,
OPP_ABILITY_OVERRIDE: Abilities.ROUGH_SKIN
```

## Checklist


Are there any localization additions or changes? If so:
